libp2p/peerinfo.nim

```diff
@@ -28,6 +33,25 @@ type
     agentVersion*: string
     privateKey*: PrivateKey
     publicKey*: PublicKey
+    signedPeerRecord*: Option[Envelope]
+
+proc createSignedPeerRecord(peerId: PeerID, addrs: seq[MultiAddress], key: PrivateKey): Result[Envelope, CryptoError] =
+  ## Creates a signed peer record for this peer:
+  ## a peer routing record according to https://github.com/libp2p/specs/blob/master/RFC/0003-routing-records.md
+  ## in a signed envelope according to https://github.com/libp2p/specs/blob/master/RFC/0002-signed-envelopes.md
+
+  # First create a peer record from the peer info
+  let peerRecord = PeerRecord.init(peerId,
+                                   getTime().toUnix().uint64, # This currently follows the recommended implementation, using unix epoch as seq no.
```

Comment:
Are you sure Moment.now wouldn't do the trick? I though it was the same thing

Reply:
Afaict there's no way to get a globally significant value from a `Moment`. `Moment`s are indeed captured using the epoch time, but that value is not exposed. The result is that `Moment`s are only meaningful as comparisons to other `Moment`s, e.g. to determine `Duration`s.

Reply:
I guess we can initialise a `Moment` at value `0` and use the `Duration` up to `Moment.now()` as `seq` number?